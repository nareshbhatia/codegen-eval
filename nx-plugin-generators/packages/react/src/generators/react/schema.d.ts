export interface ReactGeneratorSchema {
  name: string;
  tags?: string;
  directory?: string;
}
