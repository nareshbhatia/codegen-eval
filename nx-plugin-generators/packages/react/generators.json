{
  "$schema": "http://json-schema.org/schema",
  "name": "react",
  "version": "0.0.1",
  "generators": {
    "react": {
      "factory": "./src/generators/react/generator",
      "schema": "./src/generators/react/schema.json",
      "description": "react generator"
    },
    "component": {
      "factory": "./src/generators/component/generator",
      "schema": "./src/generators/component/schema.json",
      "description": "component generator"
    },
    "context": {
      "factory": "./src/generators/context/generator",
      "schema": "./src/generators/context/schema.json",
      "description": "context generator"
    }
  }
}
