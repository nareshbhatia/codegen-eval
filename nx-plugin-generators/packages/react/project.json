{
  "root": "packages/react",
  "sourceRoot": "packages/react/src",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/react",
        "main": "packages/react/src/index.ts",
        "tsConfig": "packages/react/tsconfig.lib.json",
        "assets": [
          "packages/react/*.md",
          {
            "input": "./packages/react/src",
            "glob": "**/!(*.ts)",
            "output": "./src"
          },
          {
            "input": "./packages/react/src",
            "glob": "**/*.d.ts",
            "output": "./src"
          },
          {
            "input": "./packages/react",
            "glob": "generators.json",
            "output": "."
          },
          {
            "input": "./packages/react",
            "glob": "executors.json",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/react/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/packages/react"],
      "options": {
        "jestConfig": "packages/react/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
