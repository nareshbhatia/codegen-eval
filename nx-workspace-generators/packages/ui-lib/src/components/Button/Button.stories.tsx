import * as React from 'react';
import { Story, Meta } from '@storybook/react';
import { Button } from './Button';

export default {
  title: 'Components/Button',
  component: Button,
} as Meta;

const Template: Story = (args) => <Button title={args.title} />;

export const ButtonStory = Template.bind({});
ButtonStory.storyName = 'Button';
ButtonStory.args = {
  title: 'Button',
};
