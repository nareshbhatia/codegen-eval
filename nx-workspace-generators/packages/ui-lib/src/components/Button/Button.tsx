import * as React from 'react';

interface ButtonProps {
  title: string;
}

export function Button({ title }: ButtonProps) {
  return <div>{title}</div>;
}
