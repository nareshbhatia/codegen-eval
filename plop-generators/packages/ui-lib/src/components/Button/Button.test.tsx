import { render } from '@testing-library/react';
import { Button } from './Button';

describe('<Button />', () => {
  test('variant="contained" renders correctly', () => {
    const { asFragment } = render(
      <div>
        <Button variant="contained" color="default">
          Submit
        </Button>
        <Button variant="contained" color="primary">
          Submit
        </Button>
        <Button variant="contained" color="secondary">
          Submit
        </Button>
        <Button variant="contained" disabled>
          Submit
        </Button>
      </div>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('variant="outlined" renders correctly', () => {
    const { asFragment } = render(
      <div>
        <Button variant="outlined" color="default">
          Submit
        </Button>
        <Button variant="outlined" color="primary">
          Submit
        </Button>
        <Button variant="outlined" color="secondary">
          Submit
        </Button>
        <Button variant="outlined" disabled>
          Submit
        </Button>
      </div>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('sizes render correctly', () => {
    const { asFragment } = render(
      <div>
        <Button size="small">Submit</Button>
        <Button size="medium">Submit</Button>
        <Button size="large">Submit</Button>
      </div>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('class can be added via rootClass prop', () => {
    const { asFragment } = render(<Button rootClass="ml-2">Submit</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
