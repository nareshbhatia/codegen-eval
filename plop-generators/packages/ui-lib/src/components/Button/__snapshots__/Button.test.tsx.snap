// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> class can be added via rootClass prop 1`] = `
<DocumentFragment>
  <button
    class="ml-2 button button--contained button--contained-default button--medium"
  >
    Submit
  </button>
</DocumentFragment>
`;

exports[`<Button /> sizes render correctly 1`] = `
<DocumentFragment>
  <div>
    <button
      class="button button--contained button--contained-default button--small"
    >
      Submit
    </button>
    <button
      class="button button--contained button--contained-default button--medium"
    >
      Submit
    </button>
    <button
      class="button button--contained button--contained-default button--large"
    >
      Submit
    </button>
  </div>
</DocumentFragment>
`;

exports[`<Button /> variant="contained" renders correctly 1`] = `
<DocumentFragment>
  <div>
    <button
      class="button button--contained button--contained-default button--medium"
    >
      Submit
    </button>
    <button
      class="button button--contained button--contained-primary button--medium"
    >
      Submit
    </button>
    <button
      class="button button--contained button--contained-secondary button--medium"
    >
      Submit
    </button>
    <button
      class="button button--contained button--disabled button--disabled-contained button--medium"
    >
      Submit
    </button>
  </div>
</DocumentFragment>
`;

exports[`<Button /> variant="outlined" renders correctly 1`] = `
<DocumentFragment>
  <div>
    <button
      class="button button--outlined button--outlined-default button--medium"
    >
      Submit
    </button>
    <button
      class="button button--outlined button--outlined-primary button--medium"
    >
      Submit
    </button>
    <button
      class="button button--outlined button--outlined-secondary button--medium"
    >
      Submit
    </button>
    <button
      class="button button--outlined button--disabled button--disabled-outlined button--medium"
    >
      Submit
    </button>
  </div>
</DocumentFragment>
`;
